test/integration/index.test.js

```diff
@@ -45,48 +47,93 @@ const testResponseXlsx = (res) => {
+        expect(res.statusCode).toEqual(200);
+        expect(res.body).toStrictEqual({ status: 'ok' });
+        testResponse(res);
+    });
 
-test('default mock report - XLSX - correct response', async () => {
-    const res = await createMockServer().get('/test.xlsx?currency=MMK');
-    expect(res.statusCode).toEqual(200);
-    expect(res.body).toStrictEqual({});
-    testResponseXlsx(res, { contentType: 'xlsx' });
-});
+    test('CSV - correct response', async () => {
+        const server = createMockServer({ config });
+        const watch = k8s.Watch.getInstance();
+        watch.sendResource(path.join(__dirname, 'data/test.yaml'));
+        const res = await server.get('/test?dfspId=payerfsp&currency=MMK&format=csv');
+        expect(res.statusCode).toEqual(200);
+        expect(csvParse(res.text, { columns: true })).toStrictEqual([
+            {
+                Currency: 'MMK',
+                Name: 'fsp1',
+            },
+            {
+                Currency: 'MMK',
+                Name: 'fsp3',
+            },
+        ]);
+        testResponse(res, { contentType: 'csv' });
+    });
+
+    test('XLSX - correct response', async () => {
+        const server = createMockServer({ config });
+        const watch = k8s.Watch.getInstance();
+        watch.sendResource(path.join(__dirname, 'data/test.yaml'));
+        const res = await server.get('/test?dfspId=payerfsp&currency=MMK&format=xlsx');
+        expect(res.statusCode).toEqual(200);
+        expect(res.body).toStrictEqual({});
+        testResponseXlsx(res, { contentType: 'xlsx' });
+    });
+
+    test('perms check', async () => {
+        config.oryKetoReadUrl = 'http://localhost:4466';
+        const server = createMockServer({ config });
+        const ketoMock = keto.ReadApi.getInstance();
+        const userId = 'test-user-id';
+        const fspId = 'payerfsp';
+        const reportName = 'test';
+        ketoMock.getRelationTuples = (ns, obj, rel, subj) => {
+            const result = [];
+            if (ns === 'participant' && obj === undefined && rel === 'member' && subj === userId) {
+                result.push({ object: fspId });
+            }
+            return {
+                data: {
+                    relation_tuples: result,
+                },
+            };
+        };
+        ketoMock.getCheck = (ns, obj, rel, subj) => {
+            let allowed = false;
+            if (ns === 'permission' && obj === reportName && rel === 'granted' && subj === userId) {
+                allowed = true;
+            }
+            return {
+                data: {
+                    allowed,
+                },
+            };
+        };
+        const watch = k8s.Watch.getInstance();
+        watch.sendResource(path.join(__dirname, 'data/test.yaml'));
+        let res = await server.get(`/test?dfspId=${fspId}&currency=MMK&format=xlsx`).set('x-user', userId);
+        expect(res.statusCode).toEqual(200);
+        expect(res.body).toStrictEqual({});
```

Comment:
Does this matter? Feels a bit like in this case the content of the response is distracting from the purpose of the test; so it could be removed. In other words, if this value changed, should this test now fail? Another test, sure. Perhaps not this one?